src/NotepadNext/docks/FolderAsWorkspaceDock.cpp

```diff
@@ -65,3 +74,95 @@ QString FolderAsWorkspaceDock::rootPath() const
 {
     return model->rootPath();
 }
+
+void FolderAsWorkspaceDock::onCustomContextMenu(const QPoint &point)
+{
+    QModelIndex index = ui->treeView->indexAt(point);
+    if (!index.isValid()) {
+        lastSelectedItem = model->index(rootPath());
+        ui->menuEmpty->exec(ui->treeView->viewport()->mapToGlobal(point));
+        return;
+    }
+    lastSelectedItem = index;
+    if (model->isDir(index)) {
+        ui->menuDirectory->exec(ui->treeView->viewport()->mapToGlobal(point));
+    }
+    else {
+        ui->menuFile->exec(ui->treeView->viewport()->mapToGlobal(point));
+    }
+}
+
+void FolderAsWorkspaceDock::on_actionSaveHere_triggered()
+{
+    QDir parentDir(model->filePath(lastSelectedItem));
+    auto doc = window->currentEditor();
+    QString dstName(parentDir.absoluteFilePath(doc->getName()));
+
+    if (doc->saveAs(dstName) != QFileDevice::NoError)
+    {
+        qWarning("Unable to save %s", dstName.toUtf8().constData());
+    }
+    else
+    {
+        auto newItem = model->index(dstName);
+        if (!doc->isFile()) {
+            doc->setFileInfo(dstName);
+        }
+        ui->treeView->setCurrentIndex(newItem);
+        ui->treeView->edit(newItem);
+    }
+}
+
+void FolderAsWorkspaceDock::on_actionNewFolder_triggered()
+{
+    QDir parentDir(model->filePath(lastSelectedItem));
+    int i = 1;
+
+    for (;parentDir.exists(NEW_DIR_TEMPLATE.arg(i)); i++) {
+    // Intentional
+    }
+    QString dstName = NEW_DIR_TEMPLATE.arg(i);
+
+    auto newItem = model->mkdir(lastSelectedItem, dstName);
+    if (!newItem.isValid()) {
+        qWarning("Unable to create %s", dstName.toUtf8().constData());
+    }
+    else {
+        ui->treeView->setCurrentIndex(newItem);
+        ui->treeView->edit(newItem);
+    }
+}
+
+void FolderAsWorkspaceDock::on_actionRename_triggered()
+{
+    ui->treeView->setCurrentIndex(lastSelectedItem);
+    ui->treeView->edit(lastSelectedItem);
+}
+
+void FolderAsWorkspaceDock::onFileRenamed(const QString &path, const QString &oldName, const QString &newName)
+{
+    QDir dir(path);
+    QString fileName = dir.absoluteFilePath(oldName);
+    for(auto &&editor : window->editors())
+    {
+        if (editor->isFile() && (editor->getFilePath() == fileName))
+        {
+            editor->setName(newName);
+        }
+    }
+}
+
+void FolderAsWorkspaceDock::on_actionDelete_triggered()
+{
+    QString path(model->filePath(lastSelectedItem));
+    QMessageBox::StandardButton reply = QMessageBox::question(this, tr("Delete Item"),
```

Comment:
Probably use `window->moveFileToTrash()` here.

Reply:
Implemented both options but I have no windows box nearby to test on windows.